Assets/Scripts/EOSSessionsManager.cs

```diff
@@ -1644,6 +1761,36 @@ private void OnDestroySessionCompleteCallback(ref DestroySessionCallbackInfo dat
                 return;
             }
 
+            // Before removing the session from our local data, we need to inform the owner of the session that we've left the session, if we're not the owner
+            // TODO: Validate that this gets to the members/owners of the session in time, and that we haven't already deleted the local information needed to get session information
+            Session localSession = GetSession(sessionName);
+            if (localSession != null && localSession.ActiveSession != null)
+            {
+                ActiveSessionCopyInfoOptions copyOptions = new ActiveSessionCopyInfoOptions() { };
+                Result localCopyResult = localSession.ActiveSession.CopyInfo(ref copyOptions, out ActiveSessionInfo? outActiveSessionInfo);
+                if (localCopyResult == Result.Success && outActiveSessionInfo.HasValue && outActiveSessionInfo.Value.SessionDetails.HasValue)
+                {
+                    if (outActiveSessionInfo.Value.SessionDetails.Value.OwnerUserId.Equals(EOSManager.Instance.GetProductUserId()))
+                    {
+                        // We're the owner of the session, inform everyone that it was destroyed
+                        InformSessionMembers(sessionName, P2PSessionOwnerDestroyedSessionMessageElement);
+                    }
+                    else
+                    {
+                        // Inform the owner that we've left the session
+                        InformSessionOwnerWithMessage(sessionName, P2PLeavingSessionMessageElement);
+                    }
+                }
+                else
+                {
+                    Debug.LogError($"Session Matchmaking (OnDestroySessionCompleteCallback): Failed to copy local information for session {sessionName}, so could not inform owner/members of destruction. Result code {localCopyResult}");
+                }
+            }
+            else
+            {
+                Debug.LogError($"Session Matchmaking (OnDestroySessionCompleteCallback): Could not find local Session and associated ActiveSession, so could not inform owner/members of destruction.");
+            }
+
             if (!string.IsNullOrEmpty(sessionName))
             {
                 OnSessionDestroyed(sessionName);
```

Comment:
Flattening out like this definitely looks better. I'm _slightly_ concerned about pushing down the `OnSessionDestroyed` from not happening. I wanna do all the messaging to owner/members before getting to `OnSessionDestroyed`, but I still want it to run.

That said, I'll be working on validating that this behavior always works when it should. TODO!

Trying to iterate on the above design, it feels like you've arrived at the right place. Didn't make any alterations other than writing a comment.

Reply:
I had similar thoughts regarding the `OnSessionDestroyed` call, but I wondered whether if the session name is empty or does not exist whether it makes sense to try and destroy that session or not. I suppose if it passes that test but fails early otherwise, you _would_ still want to call it, so that's a good observation.